pageserver/src/tenant.rs

```diff
@@ -3184,6 +3184,7 @@ impl Tenant {
             // Offload failures don't trip the circuit breaker, since they're cheap to retry and
             // shouldn't block compaction.
             CompactionError::Offload(_) => {}
+            e @ CompactionError::CollectKeySpaceError(_) if e.is_cancel() => {}
```

Comment:
nit: move this to the top and do `err if err.is_cancel() => {}` to handle all cancellation variants.